src/pages/api/auth/callback/[provider].ts

```diff
@@ -18,16 +18,7 @@ export default function handler(req: NextApiRequest, res: NextApiResponse) {
         return res.redirect("/login");
       }
 
-      const algo = "HS256";
-      const secret = new TextEncoder().encode(
-        process.env.TOKEN_SECRET as string,
-      );
-      const jwtToken = await new jose.SignJWT(profile)
-        .setProtectedHeader({ alg: algo })
-        .setIssuedAt()
-        .setExpirationTime("2h")
-        .sign(secret);
-
+      const jwtToken = await createToken(profile);
       res.setHeader("Set-Cookie", `jwtToken=${jwtToken}; Path=/;`);
       res.redirect("/");
```

Comment:
Add cookie expiration date, otherwise it gets deleted when the user ends the session:

```suggestion
      const jwtToken = await createToken(profile);
      const expirationDate = new Date(Date.now() + 400 * 24 * 60 * 60 * 1000); // expire in 400 days ()
      res.setHeader("Set-Cookie", `jwtToken=${jwtToken}; Path=/; Expires=${expirationDate.toUTCString()};`);
      res.redirect("/");
```